tests/templates/test_state_preparations/test_cosine_window.py

```diff
@@ -0,0 +1,112 @@
+# Copyright 2018-2021 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Unit tests for the CosineWindow template.
+"""
+# pylint: disable=too-few-public-methods
+import pytest
+import numpy as np
+import pennylane as qml
+from pennylane.wires import WireError
+
+
+class TestDecomposition:
+    """Tests that the template defines the correct decomposition."""
+
+    def test_correct_gates_single_wire(self):
+        """Test that the correct gates are applied."""
+
+        op = qml.CosineWindow(wires=[0])
+        queue = op.expand().operations
+
+        assert queue[0].name == "Hadamard"
+        assert queue[1].name == "RZ"
+        assert queue[2].name == "Adjoint(QFT)"
+        assert queue[3].name == "PhaseShift"
+
+        assert np.isclose(queue[3].data[0], np.pi / 2)
+
+    def test_correct_gates_many_wires(self):
```

Comment:
[Not blocking] you could probably combine this test with the one above using `purest.mark.parametrize`.

Reply:
(It has been left as it was)